src/Graphics/src/Graphics/Platforms/iOS/UIImageExtensions.cs

```diff
@@ -59,5 +60,23 @@ public static UIImage NormalizeOrientation(this UIImage target, bool disposeOrig
 
 			return image;
 		}
+
+		internal static UIImage ResizeImageSource(this UIImage sourceImage, nfloat maxWidth, nfloat maxHeight, CGSize originalImageSize, bool shouldScaleUp = false)
+		{
+			if (sourceImage is null || sourceImage.CGImage is null)
+				return null;
+
+			maxWidth = (nfloat)Math.Min(maxWidth, originalImageSize.Width);
+			maxHeight = (nfloat)Math.Min(maxHeight, originalImageSize.Height);
+
+			var sourceSize = sourceImage.Size;
+
+			float maxResizeFactor = (float)Math.Min(maxWidth / sourceSize.Width, maxHeight / sourceSize.Height);
+
+			if (maxResizeFactor > 1 && !shouldScaleUp)
+				return sourceImage;
+
+			return UIImage.FromImage(sourceImage.CGImage, sourceImage.CurrentScale / maxResizeFactor, sourceImage.Orientation);
```

Comment:
To avoid create a new UIImage instance, could we use `UIGraphics.BeginImageContext `to create a context for drawing the scaled image, resize it and use `UIGraphics.GetImageFromCurrentImageContext` to retrieve the resized image?

Reply:
@jsuarezruiz , I previously used this approach to resize the image. However, regarding other similar resizing issues in iOS, Mattie commented that this method may cause memory issues and suggested using the current approach I implemented for resizing. I have shared the comment link below. Please let me know if you have any concerns.

Link - https://github.com/dotnet/maui/pull/26016/files/118ad1329919d37702ffb42dc08a59edd5d9dabe#r1871770914